packages/graphql/tests/integration/issues/1221.int.test.ts

```diff
@@ -0,0 +1,694 @@
+    });
+
+    test("should return as two single 'chains', two relations deep", async () => {
+        const neoGraphql = new Neo4jGraphQL({
+            typeDefs,
+            driver,
+        });
+        schema = await neoGraphql.getSchema();
+
+        const session = driver.session();
+        try {
+            await session.run(`
+                CREATE (:${testNameDetails} { fullName: "MHA" })<-[:HAS_NAME { current: true }]-(m:${testMasterData} { current: true, id: "123" })
+                CREATE (m)<-[:ARCHITECTURE { current: true }]-(:${testSeries} { current: true, id: "321" })
+                CREATE (m)<-[:ARCHITECTURE { current: true }]-(:${testSeries} { current: true, id: "921" })
+                
+                // For verification purpose, this should be filtered out by the where clause:
+                CREATE (:${testNameDetails} { fullName: "MHDD" })<-[:HAS_NAME { current: true }]-(:${testMasterData} { current: true, id: "523" })<-[:ARCHITECTURE { current: true }]-(:${testSeries} { current: true, id: "621" })
+            `);
+        } finally {
+            await session.close();
+        }
+
+        const query = `
+                query (
+                    $where: ${testSeries}Where = { current: true }
+                    $connectionWhere: RelationPropsWhere = { current: true }
+                ) {
+                    ${testSeries.plural}(where: $where) {
+                        id
+                        architectureConnection(where: { edge: $connectionWhere }) {
+                            edges {
+                                node {
+                                    nameDetailsConnection(where: { edge: $connectionWhere }) {
+                                        edges {
+                                            node {
+                                                fullName
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            `;
+
+        const variableValues = {
+            where: {
+                current: true,
+                architectureConnection_SINGLE: {
+                    node: {
+                        nameDetailsConnection: {
+                            node: {
+                                fullName: "MHA",
+                            },
+                        },
+                    },
+                },
+            },
+            connectionWhere: {
+                current: true,
+            },
+        };
+
+        const res = await graphql({
+            schema,
+            source: query,
+            variableValues,
+            contextValue: {
+                driver,
+            },
+        });
+
+        expect(res.errors).toBeUndefined();
+
+        expect(res.data).toEqual({
+            [testSeries.plural]: expect.arrayContaining([
```

Comment:
You should either remove this `expect.arrayContaining`, or add a length check for this field to check that the length is exact.

Reply:
👍  Will add a length check